test/simulator/timestamp/src/timestamp_manager.cpp

```diff
@@ -331,3 +332,51 @@ timestamp_manager::validate_commit_timestamp(session_simulator *session, uint64_
 
     return (0);
 }
+
+/*
+ * Validate the prepare timestamp. The constraints on the prepare timestamp are:
+ * - Cannot set the prepared timestamp if the transaction is already prepared.
+ * - Cannot set prepared timestamp more than once.
+ * - Commit timestamp should not have been set before the prepare timestamp.
+ * - Prepare timestamp must be greater than the latest active read timestamp.
+ * - Prepare timestamp canot be less than the stable timestamp unless rounding
+ *   the prepare timestamp is enabled.
+ */
+int
+timestamp_manager::validate_prepare_timestamp(session_simulator *session, uint64_t prepare_ts)
+{
+    /* Cannot set the prepared timestamp if the transaction is already prepared. */
+    if (session->is_txn_prepared())
+        WT_SIM_RET_MSG(
+          EINVAL, "Cannot set the prepared timestamp if the transaction is already prepared");
+
+    /* A prepared timestamp should not have been set at this point. */
+    if (session->has_prepare_timestamp())
+        WT_SIM_RET_MSG(EINVAL, "Prepare timestamp is already set");
+
+    /* Commit timestamp should not have been set before the prepare timestamp. */
+    if (session->has_first_commit_timestamp())
+        WT_SIM_RET_MSG(
+          EINVAL, "Commit timestamp should not have been set before the prepare timestamp");
+
+    connection_simulator *conn = &connection_simulator::get_connection();
+    /* The prepare timestamp must be greater than the latest active read timestamp. */
+    uint64_t latest_active_read = conn->get_latest_active_read();
+    if (latest_active_read >= prepare_ts)
+        WT_SIM_RET_MSG(EINVAL,
+          "prepare timestamp (" + std::to_string(prepare_ts) +
+            ") must be after all active read timestamps (" + std::to_string(latest_active_read) +
+            ")");
+
+    /* Prepare timestamp canot be less than the stable timestamp unless rounding the prepare
+     * timestamp is enabled.*/
+    uint64_t stable_ts = conn->get_stable_ts();
+    if (prepare_ts <= stable_ts)
+        if (!session->is_round_prepare_ts_set())
```

Comment:
Can we use an `&&` to merge the `if` statements?

Reply:
I did that for readability.